docs/COMMANDS_AND_FLAGS.md

```diff
@@ -66,3 +67,39 @@ telegraf [flags]
 **Run telegraf with pprof:**
 
 `telegraf --config telegraf.conf --pprof-addr localhost:6060`
+
+## Secret-store management
+
+You can use telegraf to manage the secrets in the configured secret-stores.
+Please make sure you specify a config containing the secret store configuration!
+
+To list all available secret-stores run
+
+`telegraf secret list`
+
+this will print a list of all known secret-store IDs that can be used for accessing the secrets
+in that store
+
+`telegraf secret list someid`
+
+The above command will now list the *keys* of all secrets in the secret-store with ID `someid`.
+
+To access the secret *values* you can use the `get` command
+
+`telegraf secret get someid`
```

Comment:
I'll have to look into this! Thanks for looking into it. Can you maybe also show the configuration of `"secretstore"` in `telegraf.conf`?

Reply:
I see the issue now. There can be only one password per `jose` secret-store. So having two secrets with different passwords in _the same_ store does not work. That's why it is asking for a password for `/dev/shm/secrets` without appending the secret name, because it really wants the per-store password.

I'm not sure on how to document this, because this is a "feature" of the underlying library and might or might not change in the future... Any idea or suggestion?

Reply:
I see the issue. The current design is that there should be one password/secret to unlock a secret store as we have no way to specify the exact secret-key for a password. That's why there is only _one_ `password` option per secret-store.

This being said, with `jose` we have no way of testing if you are using the same password. It gets even more weird as
you can use
```toml
[[secretstores.jose]]
  id = "test"
  path = "/dev/shm/secrets"

[[secretstores.jose]]
  id = "test1"
  path = "/dev/shm/secrets"
  password = "pass1"

[[secretstores.jose]]
  id = "test2"
  path = "/dev/shm/secrets"
  password = "pass2"
```
and later reference
```toml
  username = "@{test1:name1}"
  password = "@{test2:name2}"
```

with your example and it will work.

The issue is that I have no way of finding out if _all_ secrets in the subdirectory have the same password to unlock...
Any suggestion?

Reply:
One way to handle this would be to avoid it in the first place. Eg:
* when there are no secrets in the `path`, `telegraf secret set ...` simply adds it
* when there are 1 or more secrets in the `path`, `telegraf secret set ...` first tries to unlock all of the existing, and if it can't, it errors with a useful error

Reply:
This is now completely switched to urfave CLI having single-word subcommands (`list-secretstores`, `list-secrets`, `get-secret` and `set-secret`).